public/app/percona/dbaas/components/DBCluster/EditDBClusterPage/hooks/useEditDBClusterPageDefaultValues.ts

```diff
@@ -1,26 +1,41 @@
 import { useMemo } from 'react';
-import { useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+
+import { useSelector } from 'app/types';
 
 import { getDBaaS } from '../../../../../shared/core/selectors';
 import { Kubernetes } from '../../../Kubernetes/Kubernetes.types';
-import { AddDbClusterFormValues } from '../EditDBClusterPage.types';
-import { getInitialValues } from '../EditDBClusterPage.utils';
+import { DBCluster } from '../../DBCluster.types';
+import { DB_CLUSTER_INVENTORY_URL } from '../EditDBClusterPage.constants';
+import { AddDBClusterFormValues, DBClusterPageMode, EditDBClusterFormValues } from '../EditDBClusterPage.types';
+import { getAddInitialValues, getEditInitialValues } from '../EditDBClusterPage.utils';
 
 interface EditDBClusterPageDefaultValuesProps {
   kubernetes: Kubernetes[] | undefined;
+  mode: DBClusterPageMode;
 }
 
 export const useEditDBClusterPageDefaultValues = ({
   kubernetes,
-}: EditDBClusterPageDefaultValuesProps): [AddDbClusterFormValues | undefined] => {
-  const { selectedKubernetesCluster: preSelectedKubernetesCluster } = useSelector(getDBaaS);
-
-  // TODO check of edit fields will be added in other branch
+  mode,
+}: EditDBClusterPageDefaultValuesProps): [
+  AddDBClusterFormValues | EditDBClusterFormValues | undefined,
+  DBCluster | null
+] => {
+  const history = useHistory();
+  const { selectedKubernetesCluster: preSelectedKubernetesCluster, selectedDBCluster } = useSelector(getDBaaS);
 
-  const initialValues = useMemo(
-    () => (kubernetes?.length ? getInitialValues(kubernetes, preSelectedKubernetesCluster) : undefined),
-    [kubernetes, preSelectedKubernetesCluster]
-  );
+  const initialValues = useMemo(() => {
+    if (mode === 'create') {
+      return kubernetes?.length ? getAddInitialValues(kubernetes, preSelectedKubernetesCluster) : undefined;
+    }
+    if (mode === 'edit' && selectedDBCluster) {
+      return getEditInitialValues(selectedDBCluster);
+    } else {
+      history.push(DB_CLUSTER_INVENTORY_URL);
```

Comment:
It's pretty weird to have this kind of side effect in a useMemo hook, doesn't react show some sort of an error in console?

Reply:
@matejkubinec I changed to another variant, can you look at it?